### PR TITLE
tests, Make network policy tests dual stack compatible

### PR DESCRIPTION
**Pull request by oshoval**
Check ping from all existing ips, in order
to make sure network policy is working as expected
on dual stack clusters.
    
Signed-off-by: Or Shoval <oshoval@redhat.com>

**Special notes for your reviewer:**

```release-note
None
```

### CODE REVIEW

**Comment by oshoval** [comment]
/hold 
until #4045 is merged

File: tests/networkpolicy_test.go
Comment (qinqon) [inline tests/networkpolicy_test.go:187]: If you do pass extra args to `PingFromVMConsole` it will override override the "-c 1" are you ok with that ? 

We are going to replace Eventually with "-w 8" right ?

File: tests/networkpolicy_test.go
Comment (qinqon) [inline tests/networkpolicy_test.go:185]: Do we need to check `.Interfaces).To(HaveLen(1))` ?

File: tests/networkpolicy_test.go
Comment (qinqon) [inline tests/networkpolicy_test.go:196]: We are duplicating the check also for failure we have to use Consistently more than Eventually to ensure that it does not start to work eventually.

**Review by qinqon — CHANGES_REQUESTED** [review]

File: tests/networkpolicy_test.go
Comment (oshoval) [reply tests/networkpolicy_test.go:187]: ye we dont want here eventually and we are ok to override -c 1 
we do want to see couple pings pass
we are limited by 10 seconds max unless we add the 10 seconds as a parameter

anyhow this should be discussed on 4045 i think (and indeed we discussed it there), since it belong there
https://github.com/kubevirt/kubevirt/pull/4045#discussion_r481789054
(see this please and the follows)

File: tests/networkpolicy_test.go
Comment (oshoval) [reply tests/networkpolicy_test.go:185]: i dont think so, because we have WaitUntilReady and if there isnt interface there will be a panic and the test would fail

File: tests/networkpolicy_test.go
Comment (oshoval) [reply tests/networkpolicy_test.go:196]: its not right to use Consistently here because it does ping in the beginning and then stop.

as of the 2nd Expect, its to make sure it stop was just one stop suddenly, as discussed on 4045
wdyt?

Comment (qinqon) [reply tests/networkpolicy_test.go:196]: Consistently will repeat ping until the condition does pass, I think this is what we need to use.

Comment (oshoval) [reply tests/networkpolicy_test.go:196]: both can work, but i think the English should be
"eventually ping should fail" 
because we actually know the policy takes time to be applied and at the end would be applied and then"
"eventually ping would fail"

Comment (oshoval) [reply tests/networkpolicy_test.go:196]: changed the lone Expect to Consistently on #4045 as we agreed.
This PR would be just rebased to use that, once #4045 stabilize, since this PR doesnt change this logic, just the surrounding
of it.

**Comment by oshoval** [comment]
re-rebased on newer 4045
Updated the last commit (which is the only unique of this PR) accordingly

**Review by EdDev — CHANGES_REQUESTED** [review]

File: tests/networkpolicy_test.go
Comment (oshoval) [reply tests/networkpolicy_test.go]: Done (for both)

File: tests/networkpolicy_test.go
Comment (EdDev) [inline tests/networkpolicy_test.go:212]: You just need to return the error, not add here any text. It will never be printed anyway.

File: tests/networkpolicy_test.go
Comment (EdDev) [inline tests/networkpolicy_test.go:221]: Same

File: tests/networkpolicy_test.go
Comment (oshoval) [reply tests/networkpolicy_test.go:212]: there isnt error to return, so i need to create one, 
`*new(error)` like this ?
its ugly, i guess there are better ways

Comment (oshoval) [reply tests/networkpolicy_test.go]: Done both

**Review by EdDev — APPROVED** [review]
Thank you.

**Comment by oshoval** [comment]
Rebased in order to remove dependency on merged #4045 
/hold cancel

**Review by EdDev — APPROVED** [review]

**Comment by oshoval** [comment]
both failures on 1.18 are known to be flaky and not due to this PR
/retest

**Pull request merged**
